test/suites/integration/target_group_policy_test.go

```diff
@@ -0,0 +1,130 @@
+
+		testFramework.UpdateTargetGroupPolicy(policy, &test.TargetGroupPolicyConfig{
+			Protocol: aws.String(vpclattice.TargetGroupProtocolHttps),
+		})
+
+		testFramework.ExpectUpdated(ctx, policy)
+
+		httpsTG := testFramework.GetTargetGroupWithProtocol(ctx, service, "https", "http1")
+
+		Expect(*httpsTG.Protocol).To(Equal(vpclattice.TargetGroupProtocolHttps))
+	})
+
+	It("Delete Target Group Policy reset health check config for HTTP and HTTP1 Target Group", func() {
+		policy = testFramework.CreateTargetGroupPolicy(service, &test.TargetGroupPolicyConfig{
+			PolicyName:      "test-policy",
+			Protocol:        aws.String(vpclattice.TargetGroupProtocolHttp),
+			ProtocolVersion: aws.String(vpclattice.TargetGroupProtocolVersionHttp1),
+			HealthCheck: &v1alpha1.HealthCheckConfig{
+				IntervalSeconds: aws.Int64(7),
+				StatusMatch:     aws.String("200,204"),
+			},
+		})
+
+		httpRoute = testFramework.NewHttpRoute(testGateway, service, service.Kind)
+
+		testFramework.ExpectCreated(ctx, deployment, service, policy, httpRoute)
+
+		tgSummary := testFramework.GetTargetGroup(ctx, service)
+
+		tg := testFramework.GetFullTargetGroupFromSummary(ctx, tgSummary)
+
+		Expect(*tg.Config.HealthCheck.ProtocolVersion).To(Equal(vpclattice.TargetGroupProtocolVersionHttp1))
+		Expect(*tg.Config.HealthCheck.Protocol).To(Equal(vpclattice.TargetGroupProtocolHttp))
+		Expect(*tg.Config.HealthCheck.HealthCheckIntervalSeconds).To(BeEquivalentTo(7))
+		Expect(*tg.Config.HealthCheck.Matcher.HttpCode).To(Equal("200,204"))
+
+		testFramework.ExpectDeletedThenNotFound(ctx, policy)
+
+		time.Sleep(2 * time.Second)
+
+		updatedTG := testFramework.GetFullTargetGroupFromSummary(ctx, tgSummary)
+
+		Expect(*updatedTG.Config.HealthCheck.Enabled).To(BeTrue())
+		Expect(*updatedTG.Config.HealthCheck.Path).To(Equal("/"))
+		Expect(*updatedTG.Config.HealthCheck.HealthCheckIntervalSeconds).To(BeEquivalentTo(30))
+		Expect(*updatedTG.Config.HealthCheck.HealthCheckTimeoutSeconds).To(BeEquivalentTo(5))
+		Expect(*updatedTG.Config.HealthCheck.HealthyThresholdCount).To(BeEquivalentTo(5))
+		Expect(*updatedTG.Config.HealthCheck.UnhealthyThresholdCount).To(BeEquivalentTo(2))
+		Expect(*updatedTG.Config.HealthCheck.Protocol).To(Equal(vpclattice.TargetGroupProtocolHttp))
+		Expect(*updatedTG.Config.HealthCheck.ProtocolVersion).To(Equal(vpclattice.TargetGroupProtocolVersionHttp1))
+		Expect(*updatedTG.Config.HealthCheck.Matcher.HttpCode).To(Equal("200"))
+		Expect(updatedTG.Config.HealthCheck.Port).To(BeNil())
+	})
+
+	It("Delete Target Group Policy create HTTP and HTTP1 Target Group", func() {
+		policy = testFramework.CreateTargetGroupPolicy(service, &test.TargetGroupPolicyConfig{
+			PolicyName:      "test-policy",
+			Protocol:        aws.String(vpclattice.TargetGroupProtocolHttps),
+			ProtocolVersion: aws.String(vpclattice.TargetGroupProtocolVersionHttp2),
+		})
+
+		httpRoute = testFramework.NewHttpRoute(testGateway, service, service.Kind)
+
```

Comment:
Do we really need to create httpRoute, deployment, service 3 times? can these 3 test cases reuse one  httpRoute, deployment, service to save some time?